cdap-cli/src/main/java/co/cask/cdap/cli/command/GetStreamStatsCommand.java

```diff
@@ -0,0 +1,240 @@
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.util.AbstractCommand;
+import co.cask.cdap.client.QueryClient;
+import co.cask.cdap.client.StreamClient;
+import co.cask.cdap.explore.client.ExploreExecutionResult;
+import co.cask.cdap.proto.ColumnDesc;
+import co.cask.cdap.proto.QueryResult;
+import co.cask.cdap.proto.StreamProperties;
+import co.cask.common.cli.Arguments;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multiset;
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.Inject;
+
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A CLI command for getting statistics about stream events.
+ */
+@Beta
+public class GetStreamStatsCommand extends AbstractCommand {
+
+  private static final int DEFAULT_LIMIT = 100;
+  private static final int MAX_LIMIT = 100000;
+
+  private final StreamClient streamClient;
+  private final QueryClient queryClient;
+
+  @Inject
+  public GetStreamStatsCommand(StreamClient streamClient, QueryClient queryClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.streamClient = streamClient;
+    this.queryClient = queryClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String streamId = arguments.get(ArgumentName.STREAM.toString());
+    // limit limit to [1, MAX_LIMIT]
+    int limit = Math.max(1, Math.min(MAX_LIMIT, arguments.getInt(ArgumentName.LIMIT.toString(), DEFAULT_LIMIT)));
+
+    // hack to validate streamId
+    StreamProperties config = streamClient.getConfig(streamId);
+    if (config.getFormat() == null || config.getFormat().getSchema() == null) {
+      output.printf("No schema found for stream '%s'", streamId);
+      output.println();
+      return;
+    }
+
+    output.printf("Analyzing %d stream events...", limit);
+    output.println();
+
+    // build processorMap: Hive column name -> StatsProcessor
+    Map<String, Set<StatsProcessor>> processorMap = Maps.newHashMap();
+    Schema streamSchema = config.getFormat().getSchema();
+    for (Schema.Field field : streamSchema.getFields()) {
+      Schema fieldSchema = field.getSchema();
+      String hiveColumnName = cdapSchemaColumName2HiveColumnName(streamId, field.getName());
+      processorMap.put(hiveColumnName, getProcessorsForType(fieldSchema.getType(), fieldSchema.getUnionSchemas()));
+    }
+
+    // get a list of stream events and calculates various statistics about the events
+    ListenableFuture<ExploreExecutionResult> resultsFuture = queryClient.execute(
+      "SELECT * FROM cdap_stream_" + streamId + " LIMIT " + limit);
```

Comment:
There isn't a string constant for that yet, but yes it should not be hardcoded here.